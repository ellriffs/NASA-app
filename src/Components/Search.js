import { useState } from 'react';
import getImages from '../Requests/getImages';
import "../Styles/Search.css"

const Search = () => {
    const [value, setValue] = useState();

    const handleSubmit = (event) => {

        event.preventDefault()
        getImages(value);
    };

    return(
        <>
        <form className="input-form" onSubmit={handleSubmit}>       
            <input onChange={(e) => setValue(e.target.value)} className="Search" type="text"></input>
            <button className="submit"  type="submit">Search</button>
        </form>
        </>
    );
};

export default Search;