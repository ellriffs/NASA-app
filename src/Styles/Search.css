.Search{
    display: flex;
    flex-direction: row;
    justify-content: center;
    width: 50rem;
    height: 2rem;
    font-size: 2rem;
    text-align: center;
    border-radius: 5px;
    
}

.input-form{
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.submit{
    border-radius: 5px;
    cursor: alias;
    
}
