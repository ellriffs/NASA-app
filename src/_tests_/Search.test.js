import { render } from '@testing-library/react';
import React from 'react';
import Search from '../Components/Search';

describe("Search", () => {
    const { asFragment } = render(<Search />)

    it('mounts App component to the virtual DOM', () => {
        expect(asFragment()).toMatchSnapshot();
    });
}) ;