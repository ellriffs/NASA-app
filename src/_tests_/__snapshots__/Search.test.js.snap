// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search mounts App component to the virtual DOM 1`] = `
<DocumentFragment>
  <form
    class="input-form"
  >
    <input
      class="Search"
      type="text"
    />
    <button
      class="submit"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
